src/stores/navigationStore.ts

```diff
@@ -0,0 +1,23 @@
+import { action, computed, observable } from 'mobx';
+
+export class NavigationStore {
+    @observable private _shouldShowPrompt: boolean;
```

Comment:
Maybe rename to just _showPrompt

Could also rename all the "shoulds" below

Reply:
Don't want to start booleans with a verb, otherwise you can't know if its a method or boolean